dotNETStandard/HealthVault.Mobile/cs/Xamarin.Core/ViewModels/MedicationEditViewModel.cs

```diff
@@ -58,10 +66,56 @@ public VocabularyItem Name
 
         public ICommand SaveCommand { get; }
 
-        private async Task SaveAsync(Medication medication, IThingClient thingClient, Guid recordId)
+        public override async Task OnNavigateToAsync()
+        {
+            await this.LoadAsync(async () =>
+            {
+                IVocabularyClient vocabClient = this.connection.CreateVocabularyClient();
+                var ingredientChoices = new List<VocabularyItem>();
+
+                Vocabulary ingredientVocabulary = null;
+                while (ingredientVocabulary == null || ingredientVocabulary.IsTruncated)
+                {
+                    string lastCodeValue = null;
+                    if (ingredientVocabulary != null)
+                    {
+                        if (ingredientVocabulary.Values.Count > 0)
+                        {
+                            lastCodeValue = ingredientVocabulary.Values.Last().Value;
+                        }
+                        else
+                        {
+                            break;
+                        }
+                    }
+
+                    ingredientVocabulary = await vocabClient.GetVocabularyAsync(new VocabularyKey("RxNorm Active Ingredients", "RxNorm", "09AB_091102F", lastCodeValue));
```

Comment:
I was thinking about these methods and how I *really* don't like that they only return a page of data without really giving the user any sort of obvious clue that there could be more. And with no built-in way of just getting everything. I was thinking as a start just adding "Page" to all the methods that don't give you everything at once, just to make the user aware that they need to do extra work. It might be worth the price of straying from the HV method names a little bit. We can talk more about it at standup.

---
In reply to: [117592004](https://github.com/Microsoft/healthvault-samples/pull/11#discussion_r117592004) [](ancestors = 117592004)